lib/elixir/lib/supervisor.ex

```diff
@@ -124,10 +303,54 @@ defmodule Supervisor do
       requires the whole tree to be restarted in order to
       perform such swaps.
 
-  ## Strategies
+  ## start_link/2, init/2 and strategies
+
+  So far we have started the supervisor passing a single child as a tuple
+  as well as a strategy called `:one_for_one`:
+
+      Supervisor.start_link([
+        {Stack, [:hello]}
+      ], strategy: :one_for_one)
+
+  Or:
+
+      Supervisor.init([
+        {Stack, [:hello]}
+      ], strategy: :one_for_one)
+
+  However, children can be specified in three different formats and
+  supervisors support different options. Let's formally define those.
+
+  The first argument given to `start_link/2` is a list of children which may
+  be either:
+
+      * a module - such as `Stack`. In this case, it is equivalent to passing
+        `{Stack, []}`.
```

Comment:
Why `[]` and not `nil`? I understand it needs to be `[]` for a Stack and `[]` is useful for keyword args so not against this. However it doesn't feel natural and its unclear if this is the default arg called by the supervisor or if a `child_spec/0` is defined with a default arg.